File: Remote/Source/Commands/CommandContainer.m
//
// CommandContainer.m
// iPhonto
//
// Created by Jason Cardwell on 6/29/11.
// Copyright (c) 2011 Moondeer Studios. All rights reserved.
//

#import "CommandContainer.h"

static int   ddLogLevel = DefaultDDLogLevel;

#pragma unused(ddLogLevel)

@implementation CommandContainer

/*
 * isValidKey:
 */
+ (BOOL)isValidKey:(NSString *)key {
    return NO;
}

/*
 * isValidKey:
 */
- (BOOL)isValidKey:(NSString *)key {
    return [[self class] isValidKey:key];
}

@end

File: DataModel/Model/RemoteElement/Commands/CommandSets/CommandContainer.swift
//
//  CommandContainer.swift
//  Remote
//
//  Created by Jason Cardwell on 3/2/15.
//  Copyright (c) 2015 Moondeer Studios. All rights reserved.
//

import Foundation
import CoreData
import MoonKit

@objc(CommandContainer)
public class CommandContainer: NamedModelObject {

  var containerIndex: OrderedDictionary<String, NSURL> {
    get {
      willAccessValueForKey("containerIndex")
      let containerIndex = primitiveValueForKey("containerIndex") as! OrderedDictionary<String, NSURL>
      didAccessValueForKey("containerIndex")
      return containerIndex
    }
    set {
      willChangeValueForKey("containerIndex")
      setPrimitiveValue(newValue, forKey: "containerIndex")
      didChangeValueForKey("containerIndex")
    }
  }
  @NSManaged public var buttonGroup: ButtonGroup?

  public var count: Int { return containerIndex.count }

}